drivers/platform/generic/timer.c

```diff
@@ -0,0 +1,184 @@
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include "error.h"
+#include "timer.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @breif Initialize hardware timer and the handler structure associated with
+ *        it.
+ * @param [out] desc - Pointer to the reference of the device handler.
+ * @param [in] init_param - Initialization structure.
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t timer_init(struct timer_desc **desc,
+		   struct timer_init_param *init_param)
+{
+	if(desc) {
+		// Unused variable - fix compiler warning
+	}
+	if(init_param) {
+		// Unused variable - fix compiler warning
+	}
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Free the memory allocated by timer_setup().
+ * @param [in] desc - Pointer to the device handler.
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t timer_remove(struct timer_desc *desc)
+{
+	if(desc) {
+		// Unused variable - fix compiler warning
+	}
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Start a timer.
+ * @param [in] desc - Pointer to the device handler.
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t timer_start(struct timer_desc *desc)
+{
+	if(desc) {
+		// Unused variable - fix compiler warning
+	}
+}
+
+/**
+ * @brief Stop a timer from counting.
+ * @param [in] desc - Pointer to the device handler.
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t timer_stop(struct timer_desc *desc)
+{
+	if(desc) {
+		// Unused variable - fix compiler warning
+	}
+}
+
+/**
+ * @brief Get the value of the counter register for the timer.
+ * @param [in]  desc    - Pointer to the device handler.
+ * @param [out] counter - Pointer to the counter value.
+ * @return 0 in case of success, error code otherwise.
+ */
+int32_t timer_counter_get(struct timer_desc *desc, uint32_t *counter)
+{
+	if(desc) {
+		// Unused variable - fix compiler warning
+	}
+
+	if(counter) {
+		// Unused variable - fix compiler warning
+	}
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Set the timer counter register value.
+ * @param [in] desc    - Pointer to the device handler.
+ * @param [in] new_val - The new value of the counter register.
+ * @return 0 in case of success, error code otherwise.
+ */
+int32_t timer_counter_set(struct timer_desc *desc, uint32_t new_val)
+{
+	if(desc) {
+		// Unused variable - fix compiler warning
+	}
+
+	if(new_val) {
+		// Unused variable - fix compiler warning
+	}
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Get the timer clock frequency.
+ * @param [in]  desc      - Pointer to the device handler.
+ * @param [out] count_clk - The new value of the counter register.
```

Comment:
This is a bit confusing to me, for the following reasons:

- what is the measurement unit, Hz / ticks ? It should be specified.
- If by `count_clk` you mean frequency, i think it would be more clear to name it as such ('freq', 'frequency' etc.). This comment applies also to `timer_count_clk_get` and `timer_count_clk_set`.

Reply:
You are right, I will update this.